angular/src/validators/notAllowedValueAsync.validator.ts

```diff
@@ -0,0 +1,21 @@
+import { AbstractControl, AsyncValidatorFn, ValidationErrors } from "@angular/forms";
+
+export function notAllowedValueAsync(
+  valueGetter: () => Promise<string>,
```

Comment:
Do you think it's worth the type acrobatics to allow `string | Promise<string>`?

Reply:
No, in that case you'd need to change the return value of `AsyncValidatorFn` to `AsyncValidatorFn | ValidatorFn` to and have to branch on which function to return based on the type. I think if this is needed in the future having a sync and async version of this validator would be for the best.